controller/metrics/metrics.go

```diff
@@ -229,6 +240,11 @@ func (m *MetricsServer) IncSync(app *argoappv1.Application, state *argoappv1.Ope
 	m.syncCounter.WithLabelValues(app.Namespace, app.Name, app.Spec.GetProject(), app.Spec.Destination.Server, string(state.Phase)).Inc()
 }
 
+// IncRefresh increments the refresh counter for an application
+func (m *MetricsServer) IncRefresh(app *argoappv1.Application, compareWithStr string) {
+	m.refreshCounter.WithLabelValues(app.Namespace, app.Name, app.Spec.GetProject(), app.Spec.Destination.Server, compareWithStr).Inc()
```

Comment:
We have Argo CD instances handling thousands of apps in hundreds of namespaces. Adding all those labels in the new `refreshCounter` metric will cause a huge boost in cardinality which could lead to serious issues in Prometheus. More on the subject: https://blog.fourninecloud.com/understanding-cardinality-in-prometheus-monitoring-96ad082b6398

Reply:
Hi!
After applying [this](https://github.com/argoproj/argo-cd/pull/17013) fix, I think this PR can be closed.

Reply:
Thank you for confirming.